src/theory/bags/skolem_cache.h

```diff
@@ -0,0 +1,76 @@
+/*********************                                                        */
+/*! \file skolem_cache.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief A cache of skolems for theory of sets.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__BAGS__SKOLEM_CACHE_H
+#define CVC4__THEORY__BAGS__SKOLEM_CACHE_H
+
+#include <map>
+#include <unordered_set>
+
+#include "expr/node.h"
+
+namespace CVC4 {
+namespace theory {
+namespace bags {
+
+/**
+ * A cache of all bag skolems generated by the TheoryBags class. This
+ * cache is used to ensure that duplicate skolems are not generated when
+ * possible, and helps identify what skolems were allocated in the current run.
+ */
+class SkolemCache
+{
+ public:
+  SkolemCache();
+  /** Identifiers for skolem types
```

Comment:
I would not introduce any of these methods.  It is not clear what the skolem cache of bags should do yet, or if it is even necessary.

Personally, I'd prefer if you didnt introduce this class yet at all.

Reply:
Done by removing `SkoelmCache`